### DISCUSSION TITLE
Is possible to get a smaller file to do specific action? Webp to Gif?

### DISCUSSION THREAD

**Discussion by d4annn** [ImageMagick/ImageMagick#5489] [Help]
I want to use this app but 30mb is a lot, i need something less than 5mb, i want to do this comm webp.webp -coalesce gif.gif, is possible?

**Comment by snibgo** [comment]
You can build IM with many "`./configure`" options disabled. I don't know if that will reduce it to 5MB.

**Reply by d4annn** [reply]
i followed this https://github.com/ImageMagick/ImageMagick-Windows , is this what i needed to do? or i need to build it with make file? got the configure.exe but there are no real features there

**Comment by snibgo** [comment]
That page uses Visual Studio on Windows. I don't use that. Instead, I use Cygwin tools; see [Compiling IM with Cygwin](http://im.snibgo.com/compim.htm).

**Reply by d4annn** [reply]
Im getting a little lost following that, i have alredy installed cgwin, im in the next step, what do i need to download from here? http://www.imagemagick.org/script/install-source.php there explains what i did before. Also tried the Building the Unix source step but it cant find wget command

**Comment by snibgo** [comment]
What are you asking? The page you link has a section called "Install from Windows Source".

When you have installed the source, to build with Cygwin, use five bash commands: `automake`, `autoconf`, `./configure`, `make`, `make install`.

**Reply by d4annn** [reply]
I clonned the repo, ran CloneRepositories, and opened the ImageMagick-windows-7 folder with cygdrive, i guess its there where i need to run the commands, when i run ./configure it says -bash: ./configure: No such file or directory and with the other commands , -bash: automake: command not found

**Comment by snibgo** [comment]
> i guess its there where i need to run the commands, when i run ./configure it says -bash: ./configure: No such file or directory 

If that directory does not contain a file named "`configure`" then you are in the wrong directory.

The file named "`automake`" should be in a directory that is on your system path. On my computer, it is in the directory "`C:\cygwin64\bin\`".

**Reply by d4annn** [reply]
Im trying to run it here [image](https://imgur.com/a/YJmEM8A) , it says the same 
```
Usuario@ORDENADOR /cygdrive/d/Programacion/C/ImageMagick-Windows-7/VisualMagick/configure
$ ./configure --with-bzlib=false
-bash: ./configure: No such file or directory
```
in my cygwin64 bin folder there is not automake or other commands like make/autoconf... do i need to install them by my own? or they got automatically downloaded when i installed cygwin?

**Comment by snibgo** [comment]
> $ ./configure --with-bzlib=false
-bash: ./configure: No such file or directory

As I said, you are in the wrong directory.

I linked to http://im.snibgo.com/compim.htm . On that page, I said:
> I installed the complete binary Cygwin "Devel" (development) package, to ensure I had make, the compiler and linker and so on. 

Did you download the Cygwin "Devel" (development) package?

**Reply by d4annn** [reply]
Now i have installed everything well, i think im in the correct directory [image](https://imgur.com/a/tYiqjI5) but when i run the example command it gives me 
```
Usuario@ORDENADOR /cygdrive/d/Programacion/C/ImageMagick-Windows-7/ImageMagick
$ ./configure --prefix=/home/Alan/iminst --with-quantum-depth=32 --enable-hdri --with-jp2=yes --with-rsvg=yes --with-windows-font-dir=/cygdrive/c/Windows/Fonts
./configure: line 17: $'\r': command not found
./configure: line 34: syntax error near unexpected token `newline'
'/configure: line 34: `     ;;
```
and i dont understand the imconfg.sh thing

**Comment by snibgo** [comment]
> ./configure: line 17: $'\r': command not found
./configure: line 34: syntax error near unexpected token `newline'

See [IM with Cygwin: Line ends](http://im.snibgo.com/cygwin.htm#lineends)

My `./configure` line is quite long. Rather than copying it from somewhere each time I build, I put the long command in a shell script called `imconfig.sh`.

**Comment by snibgo** [comment]
In bash, `echo $SHELLOPTS` will show you the current value of SHELLOPTS. Does it contain "`igncr`"? If not, then you need to set it in Windows. See the instructions on my page at "Setting environment variables".

**Reply by d4annn** [reply]
Got this when trying to run any ./configure
```
$ ./configure --with-bzlib=no
checking build system type... x86_64-pc-cygwin
checking host system type... x86_64-pc-cygwin
checking target system type... x86_64-pc-cygwin
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '197609' is supported by ustar format... yes
[config.log](https://github.com/ImageMagick/ImageMagick/files/9462306/config.log)

checking whether GID '197609' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether make supports nested variables... (cached) yes
Configuring ImageMagick 7.1.0-48
checking whether build environment is sane... yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... no
checking for cc... no
checking for cl.exe... no
checking for clang... no
configure: error: in `/cygdrive/d/Programacion/C/ImageMagick-Windows-7/ImageMagick':
configure: error: no acceptable C compiler found in $PATH
```
found many fixes but for Ubuntu `yum install gcc`

**Comment by snibgo** [comment]
It looks as if you have successfully installed bash and some tools (eg gawk) but not a compiler.

Did `automake` and `autoconf` work without error?

If they did, then try `where gcc` and `whereis gcc`. For these, I get:
```
f:\prose\PICTURES>where gcc
C:\cygwin64\bin\gcc.exe

f:\prose\PICTURES>whereis gcc
gcc: /usr/bin/gcc.exe /usr/lib/gcc /usr/share/man/man1/gcc.1.gz /usr/share/info/gcc.info.gz
```
If you have gcc, try `gcc --version`. I get:
```
f:\web\im>gcc --version
gcc (GCC) 11.2.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

If your gcc can't be found, then check your Cygwin setup. Did you install the complete binary Cygwin "Devel" (development) package? Did that include gcc?

**Reply by d4annn** [reply]
autoconf gives `/cygdrive/c/opt/autoconf: line 615: c:/progra~1/autoconf/bin/autom4te: No such file or directory` and automake command doesnt exist for me, i installed all packages i dont know why i dont have them

**Comment by snibgo** [comment]
> ... and automake command doesnt exist for me, ...

So you haven't successfully installed the "Devel" packages. On my Windows 8.1 laptop, the Windows directory `C:\cygwin64\bin\` contains files `automake`, `autoconf`, `gcc.exe`, `make.exe` and many others. And `C:\cygwin64\bin\` is on my system path, of course.

To check the path, use this Windows command:
```
f:\web\im>set path

Path=C:\cygwin64\bin;C:\cygwin64\usr\local\bin;F:\pictures\bash;C:\Program Files
 (x86)\HP SimplePass\x64;C:\Program Files (x86)\HP SimplePass\;C:\Program Files
(x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Program Files (x
86)\NVIDIA Corporation\PhysX\Common;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\Sy
... and so on.
```

**Reply by d4annn** [reply]
Now everything is installed correctly, when i write `./configre --with-bzlib=no` gives me 
```
./configure: 35: Syntax error: newline unexpected (expecting ")")
```
in the cygwin `echo $SHELLOPTS` doesnt contain igncr but in the windows prompt `set SHELLOPTS` returns me `SHELLOPTS=igncr`

**Comment by snibgo** [comment]
> in the cygwin echo $SHELLOPTS doesnt contain igncr but in the windows prompt set SHELLOPTS returns me SHELLOPTS=igncr

I don't know how that can happen.

**Reply by d4annn** [reply]
Is there any way to change the configuration by hand? and not by commands?
